mapping_parenting_tech/analysis/prototyping/play_store/get_play_store_apps.py

```diff
@@ -7,141 +7,413 @@
+    """
 
-re_pattern = r"(?<=href=\"\/store\/apps\/details\?id=)(.*?)(?=\")"
-link_targets = re.findall(re_pattern, html)
-app_ids = list(dict.fromkeys(link_targets))
-output_ids = {"Parenting apps": app_ids}
-logging.info("Apps ids parsed ok")
+    output_target = PROJECT_DIR / "outputs/data/" / filename
+
+    with open(output_target, "wb") as output_file:
+        pickle.dump(app_list, output_file)
+
+    # if we got here, it's all ok so return True
+    return True
 
-# %% [markdown]
-# Save app ids to JSON file
 
 # %%
-output_target = OUTPUT_PATH / OUTPUT_PLAY_IDS_FILE
-with open(output_target, "w") as output_file:
-    json.dump(output_ids, output_file)
-output_file.close
-print("App ids saved to", output_target)  # logging.info throws an error here
+save_app_ids(
+    get_playstore_app_ids("play_store/parenting_top_grossing.html"),
+    "parenting_top_grossing_ids.pickle",
+)
+
+
+# %%
+def load_app_ids(filename: str) -> list:
+    """
+    Loads a pickled list of app ids from a given file and returns the list object
+
+    Arguments:
+        filename: str - file name of pickle file to load within PARENT_DIR
+    """
+
+    app_id_list = []
+    with open(PROJECT_DIR / filename, "rb") as p_list:
+        app_id_list = pickle.load(p_list)
+
+    return app_id_list
+
 
 # %% [markdown]
-# Retrieve app details
+# ## Retrieve app details
 
 # %%
-all_app_details = dict()
-remove_apps = list()
-review_counter = 0
-most_reviews = {"id": 0, "count": 0}
-
-for app_id in tqdm(output_ids["Parenting apps"], desc="Retrieving app details"):
-    try:
-        app_details = app(
-            app_id, lang="en", country="gb"  # defaults to 'en'  # defaults to 'us'
-        )
-        all_app_details.update({app_id: app_details})
-        review_counter += app_details["reviews"]
-        if app_details["reviews"] > most_reviews["count"]:
-            most_reviews["id"] = app_id
-            most_reviews["count"] = app_details["reviews"]
+def get_playstore_app_details(app_id_list: list):
+    """
+    Returns dict of app details
+    """
 
-    except Exception as e:  # needs modifying to capture specific errors
-        logging.info(f"Error on app id {app_id}: {e} {repr(e)}")
-        remove_apps.append(app_id)
+    all_app_details = dict()
+    remove_apps = list()
 
-for app_id in remove_apps:
-    output_ids["Parenting apps"].remove(app_id)
+    for app_id in tqdm(
+        app_id_list,
+        desc="Retrieving app details",
+    ):
+        try:
+            app_details = app(
+                app_id, lang="en", country="gb"  # defaults to 'en'  # defaults to 'us'
+            )
+            all_app_details.update({app_id: app_details})
+
+        except Exception as e:  # needs modifying to capture specific errors
+            logging.info(f"Error on app id {app_id}: {e} {repr(e)}")
+            remove_apps.append(app_id)
+
+    for app_id in remove_apps:
+        all_app_details.remove(app_id)
```

Comment:
This line caused an error `AttributeError: 'dict' object has no attribute 'remove'` Is it working for you?
Here is an alternative way that works for me:
```suggestion
        all_app_details.pop(app_id, None)
```

Reply:
Odd, given that `all_app_details` is defined as a list, but have changed per suggestion